components/infobox/wikis/freefire/infobox_person_custom.lua

```diff
@@ -0,0 +1,175 @@
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Player = require('Module:Infobox/Person')
+local String = require('Module:StringUtils')
+local Class = require('Module:Class')
+local Page = require('Module:Page')
+local TeamHistoryAuto = require('Module:TeamHistoryAuto')
+local Region = require('Module:Region')
+local Variables = require('Module:Variables')
+local Template = require('Module:Template')
+
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Title = require('Module:Infobox/Widget/Title')
+local Center = require('Module:Infobox/Widget/Center')
+
+local _EMPTY_AUTO_HISTORY = '<table style="width:100%;text-align:left"></table>'
+
+local _BANNED = {['banned'] = {category = 'Banned players', variable = 'Banned', isplayer = true},}
+
+local _ROLES = {
+	-- Players
+	['support'] = {category = 'Support players', variable = 'Support', isplayer = true},
+	['rusher'] = {category = 'Rusher', variable = 'Rusher', isplayer = true},
+	['sniper'] = {category = 'Snipers', variable = 'Snipers', isplayer = true},
+	['granader'] = {category = 'Granader', variable = 'Granader', isplayer = true},
+	['igl'] = {category = 'In-game leaders', variable = 'In-game leader', isplayer = true},
+
+	--Staff and Talents
+	['analyst'] = {category = 'Analysts', variable = 'Analyst', talent = true},
+	['host'] = {category = 'Hosts', variable = 'Host', talent = true},
+	['journalist'] = {category = 'Journalists', variable = 'Journalist', talent = true},
+	['coach'] = {category = 'Coaches', variable = 'Coach', staff= true},
+	['caster'] = {category = 'Casters', variable = 'Caster', talent = true},
+	['manager'] = {category = 'Managers', variable = 'Manager', staff = true},
+	['producer'] = {category = 'Producers', variable = 'Producer', talent = true},
+	['streamer'] = {category = 'Streamers', variable = 'Streamer', talent = true},
+}
+
+local CustomPlayer = Class.new()
+local CustomInjector = Class.new(Injector)
+
+local _args
+
+function CustomPlayer.run(frame)
+	local player = Player(frame)
+	_args = player.args
+
+	player.defineCustomPageVariables = CustomPlayer.defineCustomPageVariables
+	player.getPersonType = CustomPlayer.getPersonType
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'history' then
+		local manualHistory = _args.history
+		local automatedHistory = TeamHistoryAuto._results({
+			convertrole = 'true',
+			player = _pagename
+		}) or ''
+		automatedHistory = tostring(automatedHistory)
+		if automatedHistory == _EMPTY_AUTO_HISTORY then
+			automatedHistory = nil
+		end
+
+	if not (String.isEmpty(manualHistory) and String.isEmpty(automatedHistory)) then
+			return {
+				Title{name = 'History'}, Center{content = {manualHistory}}, Center{content = {automatedHistory}},
```

Comment:
for readability it might be good to do 1 line per widget, i.e.
```suggestion
				Title{name = 'History'},
				Center{content = {manualHistory}},
				Center{content = {automatedHistory}},
```